tests/std/tests/P0896R4_counted_iterator_death/test.cpp

```diff
@@ -0,0 +1,248 @@
+    counted_iterator<int*> cit{globalArray, 2};
+    (void) (3 + cit); // cannot seek counted_iterator after end
+}
+
+void test_case_operator_retreat_before_begin() {
+    counted_iterator<vit> cit{begin(checkedArray), 5};
+    cit -= 1; // cannot seek counted_iterator before begin
+}
+
+void test_case_operator_retreat_negative_after_end() {
+    counted_iterator<int*> cit{globalArray, 0};
+    cit -= -1; // cannot seek counted_iterator after end
+}
+
+void test_case_operator_retreat_copy_before_begin() {
+    counted_iterator<vit> cit{begin(checkedArray), 5};
+    (void) (cit - 1); // cannot seek counted_iterator before begin
+}
+
+void test_case_operator_retreat_copy_after_end() {
+    counted_iterator<int*> cit{globalArray, 0};
+    (void) (cit - -1); // cannot seek counted_iterator after end
+}
+
+void test_case_operator_subtract_incompatible_different_data() {
+    counted_iterator<int*> cit1(globalArray, 2);
+    counted_iterator<int*> cit2(otherArray, 2);
+    (void) (cit1 - cit2); // cannot subtract incompatible counted_iterators
+}
+
+void test_case_operator_subtract_incompatible_different_size() {
+    counted_iterator<int*> cit1(globalArray, 3);
+    counted_iterator<int*> cit2(globalArray, 4);
+    (void) (cit1 - cit2); // cannot subtract incompatible counted_iterators
+}
+
+void test_case_operator_subtract_incompatible_value_initialized() {
+    counted_iterator<int*> cit{globalArray, 0};
+    (void) (cit - counted_iterator<int*>{}); // cannot subtract incompatible counted_iterators
+}
+
+void test_case_operator_equal_incompatible_different_data() {
+    counted_iterator<int*> cit1(globalArray, 2);
+    counted_iterator<int*> cit2(otherArray, 2);
+    (void) (cit1 == cit2); // cannot compare incompatible counted_iterators for equality
+}
+
+void test_case_operator_equal_incompatible_different_size() {
+    counted_iterator<int*> cit1(globalArray, 3);
+    counted_iterator<int*> cit2(globalArray, 4);
+    (void) (cit1 == cit2); // cannot compare incompatible counted_iterators for equality
+}
+
+void test_case_operator_equal_incompatible_value_initialized() {
+    counted_iterator<int*> cit{globalArray, 0};
+    (void) (cit == counted_iterator<int*>{}); // cannot compare incompatible counted_iterators for equality
+}
+
+void test_case_operator_spaceship_incompatible_different_data() {
+    counted_iterator<int*> cit1(globalArray, 3);
+    counted_iterator<int*> cit2(otherArray, 3);
+    (void) (cit1 <=> cit2); // cannot compare incompatible counted_iterators
+}
+
+void test_case_operator_spaceship_incompatible_different_size() {
+    counted_iterator<int*> cit1(globalArray, 3);
+    counted_iterator<int*> cit2(globalArray, 4);
+    (void) (cit1 <=> cit2); // cannot compare incompatible counted_iterators
+}
+
+void test_case_operator_spaceship_incompatible_value_initialized() {
+    counted_iterator<int*> cit{globalArray, 0};
+    (void) (cit <=> counted_iterator<int*>{}); // cannot compare incompatible counted_iterators
+}

```

Comment:
Similarly, we should have positive tests that value-init can be compared and differenced with another value-init, with the answers "same" and "zero".

Reply:
Here I am unsure as the comparison operators have the "same sequence" preconditition.

This would essentially amount to "nullptr == nullptr" so meh

Reply:
Added more tests